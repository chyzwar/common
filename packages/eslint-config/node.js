import eslint from "@eslint/js";
import tsEslint from "typescript-eslint";
import typescript from "./rules/typescript.js";
import jest from "./rules/jest.js";
import esm from "./rules/esm.js";
import globals from "globals";

export default tsEslint.config(
  eslint.configs.recommended,
  ...tsEslint.configs.all,
  ...tsEslint.configs.strictTypeChecked, 
  ...typescript,
  ...jest,
  ...esm, 
  {
    files: [
      "**/*.js", 
      "**/*.cjs",
      "**/*.mjs",
      
      "**/drizzle.config.ts",
      "**/runner.config.ts",
      "**/vite.config.ts",
      "**/vitest.setup.ts",
      "**/vitest.config.ts",

      "deployer.config.ts",
      "vitest.workspace.ts",
    ],
    ...tsEslint.configs.disableTypeChecked,
  },
  {
    languageOptions: {
      globals: {
        ...globals.node,
      },
    },
  }
);