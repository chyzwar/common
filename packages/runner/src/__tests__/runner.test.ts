import { expect, describe, it, vi } from "vitest";
import { dirname, resolve } from "node:path";
import { fileURLToPath } from "node:url";

vi.mock("node:process", () => {
  const currentDir = dirname(fileURLToPath(import.meta.url));

  return {
    cwd: (): string => resolve(currentDir, "../__fixtures__"),
    argv: [
      "node",
      "runner",
      "test",
    ],
  };
});

vi.mock("../Logger.js", async () => {
  const mock = await import("../__mocks__/Logger.js");
  return mock;
});

const Logger = await import("../Logger.js");

describe("runner", () => {
  it("should load configuration from runner.config.js", async () => {
    await import("../runner.js");
    await vi.dynamicImportSettled();
    expect(Logger.default.calls).toMatchSnapshot();
  });
});
