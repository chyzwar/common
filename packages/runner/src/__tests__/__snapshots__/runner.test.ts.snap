// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`runner > should load configuration from runner.config.js 1`] = `
[
  [
    "Completed tasks: test in ",
  ],
  [
    "Failed with error: Error: Missing tasks definition for: test",
  ],
]
`;
